source/objects/Map.cpp

```diff
@@ -199,99 +204,74 @@ bool Map::init(const std::shared_ptr<AssetManager> &assets,
     _json = json;
 
     // Initial geometry
-    float w = json->get("width")->asFloat();
-    float h = json->get("height")->asFloat();
-    _bounds.size.set(w, h);
-    
-    _groundnode = ShaderNode::alloc(renderer, ShaderNode::ShaderType::GROUND);
-    _wheatnode = ShaderNode::alloc(renderer, ShaderNode::ShaderType::WHEAT);
-    _groundnode->setPriority(float(Map::DrawOrder::GROUND));
-    _wheatnode->setPriority(float(Map::DrawOrder::WHEAT));
+    _bounds.size.set(_json->getFloat("width"), _json->getFloat("height"));
 
     setRootNode(root);
 
     return true;
 }
 
-bool Map::populate() {
-
+void Map::populate() {
+    
     /** Create the physics world */
     _world = physics2::net::NetWorld::alloc(getBounds(), Vec2(0, 0));
-
-    _farmerPlaying = false;
-
-    auto carrots = _json->get("carrots");
-    if (carrots != nullptr) {
-        // Convert the object to an array so we can see keys and values
-        int csize = (int) carrots->size();
-        for (int ii = 0; ii < csize; ii++) {
-            loadCarrot(carrots->get(ii));
-        }
-    } else {
-        CUAssertLog(false, "Failed to load carrots");
-    }
-
-    auto babies = _json->get("babies");
-    if (babies != nullptr) {
-        // Convert the object to an array so we can see keys and values
-        int bsize = (int) babies->size();
-        for (int ii = 0; ii < bsize; ii++) {
-            loadBabyCarrot(babies->get(ii));
-        }
-    } else {
-        CUAssertLog(false, "Failed to load babies");
-    }
-
-    auto farmers = _json->get("farmers");
-    if (farmers != nullptr) {
-        // Convert the object to an array so we can see keys and values
-        int fsize = (int) farmers->size();
-        for (int ii = 0; ii < fsize; ii++) {
-            loadFarmer(farmers->get(ii));
-        }
-    } else {
-        CUAssertLog(false, "Failed to load farmers");
-    }
-
-//    auto wheat = _json->get("wheat");
-//    if (wheat != nullptr) {
-//        // Convert the object to an array so we can see keys and values
-//        int wsize = (int) wheat->size();
-//        for (int ii = 0; ii < wsize; ii++) {
-//            loadWheat(wheat->get(ii));
-//        }
-//    } else {
-//        CUAssertLog(false, "Failed to load wheat");
-//    }
     
-    auto plantingSpots = _json->get("plantingSpots");
-    if(plantingSpots != nullptr) {
-        // Convert the object to an array so we can see keys and values
-        int wsize = (int) plantingSpots->size();
-        for (int ii = 0; ii < wsize; ii++) {
-            loadPlantingSpot(plantingSpots->get(ii));
-        }
-    } else {
-        CUAssertLog(false, "Failed to load planting spots");
-    }
+    int     width = _json->getFloat("width");
```

Comment:
getting a warning for line 219 that this variable is unused

Reply:
The level-editor folder is there since it also contains the tileset files for running Tiled as well as the .tiled-project file (which I think is necessary for others to use it). I have it in the assets folder so it can actually show up in XCode for convenience purposes. I will add a README for using the level-editor.

I have also seen the baby carrot issue on occasion, it seems like it only happens once in a while, I'm assuming it is related to the fact that currently all players are moving baby carrots with their AIController instead of just the host.